### PR TITLE
ethclient: better test suite for ethclient package

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
This PR extends the ethclient test suite and increases code coverage of the ethclient
package from ~15% to >55%. These tests act as early smoke tests to signal issues in the
RPC-interface. E.g. if a functionality like eth_chainId or eth_call breaks, the test
will break.

### CODE REVIEW

**Review by DGKSK8LIFE — APPROVED** [review]
Pretty elegant solution. LGTM

**Review by holiman — APPROVED** [review]
LGTM (but the tests run sequentially, with some changes they could be made to run in parallel)

**Comment by DGKSK8LIFE** [comment]
> LGTM (but the tests run sequentially, with some changes they could be made to run in parallel)

Do you mean concurrently? @holiman

**Comment by holiman** [comment]
Yes. The `t.Parallel()` means `TestEthClient` can (will be ) run concurrently with other tests in the same package, but the test itself executes 8 subtests sequentially. If `t.pParallel()` is invoked once for each, they too will execute concurrently. Don't know if it makes any difference, depends on how fast they are already.

**Comment by MariusVanDerWijden** [comment]
They run (all) in 0.3 seconds, so it's not really an issue

**Comment by DGKSK8LIFE** [comment]
> They run (all) in 0.3 seconds, so it's not really an issue

Sure, but what about when the testing codebase scales up? You need to accommodate for said scale while maintaining the same or better performance.

**Comment by holiman** [comment]
The full suite takes `26m` on appveyor, shaving subseconds isn't a priority, at least not if it means making the code uglier

**Comment by DGKSK8LIFE** [comment]
> The full suite takes 26m on appveyor, shaving subseconds isn't a priority, at least not if it means making the code uglier

Agreed.

File: ethclient/ethclient_test.go [hunk 1]
```diff
@@ -353,18 +387,184 @@ func TestChainID(t *testing.T) {
...
+		return err
+	}
+	// Create transaction
+	tx := types.NewTransaction(0, common.Address{1}, big.NewInt(1), 22000, big.NewInt(1), nil)
+	signer := types.NewEIP155Signer(chainID)
+	signature, err := crypto.Sign(signer.Hash(tx).Bytes(), testKey)
+	if err != nil {
+		return err
+	}
+	signedTx, err := tx.WithSignature(signer, signature)
```
Comment (MariusVanDerWijden) [inline ethclient/ethclient_test.go:567]: ineffassign

**Review by DGKSK8LIFE — APPROVED** [review]

**Pull request merged**